src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -53,17 +65,24 @@ def invalid_metric_tags(tags: Mapping[str, str]) -> Sequence[str]:
     return invalid_strs
 
 
+class InboundMessage(TypedDict):
+    # Note: This is only the subset of fields we access in this file.
+    org_id: int
+    name: str
+    type: str
+    tags: Mapping[str, str]
+
+
 class IndexerBatch:
     def __init__(self, use_case_id: UseCaseKey, outer_message: Message[MessageBatch]) -> None:
         self.use_case_id = use_case_id
         self.outer_message = outer_message
+        self.cardinality_limiter_state: Optional[CardinalityLimiterState[PartitionIdxOffset]] = None
 
-    @metrics.wraps("process_messages.parse_outer_message")
-    def extract_strings(self) -> Mapping[int, Set[str]]:
-        org_strings = defaultdict(set)
-
+    @metrics.wraps("process_messages.extract_messages")
+    def extract_messages(self) -> None:
```

Comment:
Since now this is the method that initializes the state of the batch.
Let's move it into the initializer (or create a static factory method if you prefer). 
So we remove the sequential coupling: init, then having to call `extract_messages` to do something correct. This does not return anything anymore so it is just initializing the valid state for this class.

Reply:
I made the method private and call it from init. Moving the code itself looked slightly uglier if i wanted to preserve the timer metric.